traffic_ops/testing/api_contract/v4/conftest.py

```diff
@@ -0,0 +1,103 @@
+"""This module is used to create a Traffic Ops session 
+and to store prerequisite data for endpoints"""
+import json
+import logging
+import sys
+from random import randint
+from urllib.parse import urlparse
+import pytest
+from trafficops.tosession import TOSession
+from trafficops.restapi import OperationError
+
+
+# Create and configure logger
+logger = logging.getLogger()
+
+
+def pytest_addoption(parser):
+    """Passing in Traffic Ops Arguments [Username, Password, Url and Hostname] from Command Line"""
+    parser.addoption(
+        '--to_user', action='store', help='User name for Traffic Ops Session'
+    )
+    parser.addoption(
+        '--to_password', action='store', help='Password for Traffic Ops Session'
+    )
+    parser.addoption(
+        '--to_url', action='store', help='Traffic Ops URL'
+    )
+
+
+@pytest.fixture(name="to_args")
+def to_data(pytestconfig):
+    """PyTest fixture to store Traffic ops Arguments passed from command line"""
+    args = {}
+    with open('to_data.json', encoding="utf-8", mode='r') as session_file:
+        data = json.load(session_file)
+    session_data = data["test"]
+    args['api_version'] = session_data.get("api_version")
+    args['port'] = session_data.get("port")
+
+    to_user = pytestconfig.getoption('--to_user')
+    to_password = pytestconfig.getoption('--to_password')
+    to_url = pytestconfig.getoption('--to_url')
+
+    if not all([to_user, to_password, to_url]):
+        logger.info(
+            "Traffic Ops session data were not passed from Command line Args")
+        args['user'] = session_data.get('user')
+        args['password'] = session_data.get('password')
+        args['url'] = session_data.get('url')
+    else:
+        args['user'] = to_user
+        args['password'] = to_password
+        args['url'] = to_url
+        logger.info("Parsed Traffic ops session data from args %s", args)
+    return args
+
+
+@pytest.fixture(name="to_session")
+def to_login(to_args):
```

Comment:
missing argument type and function return type

Reply:
Added argument type and function return type.